### PR TITLE
Chunk time ranges are inclusive.

### LINKED ISSUE #1522: Test flake in TestChunkStoreRandom
Failed at:
https://circleci.com/gh/cortexproject/cortex/7713
https://circleci.com/gh/cortexproject/cortex/7830
https://circleci.com/gh/cortexproject/cortex/8092

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #1522 "ChunkStoreRandom Flake".

I believe this was a real bug - queries that ended on exact day boundaries wouldn't query chunks which started on exact day boundaries.

### CODE REVIEW

File: pkg/chunk/chunk_store_test.go [hunk 1]
```diff
@@ -602,8 +602,8 @@ func TestChunkStoreRandom(t *testing.T) {
 
 			// pick two random numbers and do a query
 			for i := 0; i < 100; i++ {
-				start := rand.Int63n(100 * chunkLen)
-				end := start + 1 + rand.Int63n((100*chunkLen)-start)
+				start := rand.Int63n(99 * chunkLen)
```
Comment (bboreham) [inline pkg/chunk/chunk_store_test.go]: why do we want this change?

Comment (tomwilkie) [reply pkg/chunk/chunk_store_test.go]: At the end of the list of chunks there are no more chunks.  It simplifies the test logic below to ensuring we never query over the end of the list of chunks.

**Comment by tomwilkie** [comment]
> Have you tested that tablemanager still works as expected? ISTR the "If through ends on 00:00 of the day, don't include the upcoming day" check is important there.

Not specifically, but we've been running this for ~3 weeks and not found any problems.

**Comment by bboreham** [comment]
Turns out I was thinking of a very similar line in `PeriodicTableConfig.periodicTables()` added [here](https://github.com/cortexproject/cortex/commit/e302396922e85d75507327d21c2e461933008d99#diff-017f9527646e606d73ab98c941d602cdR366).

LGTM

**Pull request merged**